### ISSUE TITLE
Victory Pie labels don't update more than once

### ISSUE THREAD

**Issue by nicotroia** [FormidableLabs/victory-native#142]
Dynamic labels don't update more than once, when calculated in a function. 

Example
```
<VictoryPie
  data={this.state.weeklyAverage}
  x='day'
  y='cups'
  events={[{
    target: "data",
    eventHandlers: {
      onPressIn: (event) => {
        this.setState({pieNumberMode: !this.state.pieNumberMode})
      }
    }
  }]}
  labels={(d) => {
    return this.state.pieNumberMode ? d.clickLabel : d.day
    // return Math.random().toFixed(2) // same as above, only changes once
  }}
/>
```

I also tried using a mutation within the click event, but I'm not able to re-fire the click event to generate updated labels when necessary (some state change). OR, I'd also be happy with being able to reset the labels, but again, the `labels` functions does not seem to actually re-render the labels when a state changes

Running "victory-native": "0.14.0"

**Comment by nicotroia** [comment]
Edit: you can pass a random prop to the `<VictoryPie/>` itself (like other react-native components) to force a re-render on state change. This workaround is good enough for me.

```
<VictoryPie
    thing={this.state.pieNumberMode}
    ...
```

**Comment by ohmygodntt** [comment]
Hi nicotroia
Did you find out solution for your issue? I have same your issue :(

**Comment by ssg-luke** [comment]
I'm experiencing a similar issue with Victory Pie. I am using `labelComponent` with a custom component that returns a `<VictoryLabel>`. If a certain prop to that component changes it is supposed to render the label in a different position by changing the `x` and `y` props of the label. However nothing happens to the label. I also tried changing `dx` and `dy` to no effect. Changing these props with hot reloading also has no effect.

**Comment by boygirl** [comment]
Victory's `shouldComponentUpdate` logic involved an equality check that would return `true` if both values were functions. New sCU logic will return false when functions are non-identical. This issue should be fixed in `0.18.0`
